src/anndata/__init__.py

```diff
@@ -97,3 +104,15 @@ def read(*args, **kwargs):
     "experimental",
     "settings",
 ]
+
+
+class CustomExperimental(types.ModuleType):
+    def __getattribute__(self, key: str) -> Any:
```

Comment:
why not just define `def __getattr__(key: str) -> Any: ...` in the experimental module directly?

https://peps.python.org/pep-0562/ exists since 3.7

Reply:
This has the advantage of never having to thinking about this again.  As we move things out of experimental into the public API, we'll never have to update this.

Reply:
Maybe you could do `__getattr__` without a circular import? I guess I can try but I just kind of assumed not

Reply:
Although now that I think about it, maybe a blanket warning isn't such a good idea...would want to be explicit about deprecations

Reply:
You can just import inside of `__getattr__`, no?

`experimental/__init__.py`:
```py
[…]

def __getattr__(key: str) -> Any:
    import anndata

    if key in anndata.__all__: ...
```